setup.py

```diff
@@ -81,7 +82,9 @@ def get_version(rel_path):
             "mypy<=0.982",
             "pytest",
             "pytype",
+            "redis",
```

Comment:
Do you want to also create separate extra_require keys for specific backends? For example, if someone is using the redis backend they can install using something like `pip install tensorflow-similarity[redis,tensorflow]`

Reply:
We can do that, the part we need to decide about though is how to handle conditional imports as to not break the code when those packages are not available. Should we try import except, and then not import modules that use those libraries on import errors (it will be weird given we will have to import those libraries where they are not needed only to see if they exist).
Or do we go with multiple releases